packages/screens/LaunchpadERC20/LaunchpadERC20Sales/LaunchpadERC20CreateSaleForm.tsx

```diff
@@ -0,0 +1,190 @@
+import { zodResolver } from "@hookform/resolvers/zod";
+import React, { useEffect } from "react";
+import { useForm } from "react-hook-form";
+import { TouchableOpacity, View } from "react-native";
+
+import { LaunchpadERC20CreateSaleFooter } from "./LaunchpadERC20CreateSaleFooter";
+import { useCreateSaleState, zodCreateSaleForm } from "../hooks/useCreateSale";
+
+import { BrandText } from "@/components/BrandText";
+import ToggleButton from "@/components/buttons/ToggleButton";
+import { TextInputCustom } from "@/components/inputs/TextInputCustom";
+import { SpacerColumn } from "@/components/spacer";
+import useSelectedWallet from "@/hooks/useSelectedWallet";
+import { neutral77 } from "@/utils/style/colors";
+import { fontSemibold14, fontSemibold20 } from "@/utils/style/fonts";
+
+// TODO: ADD A WAY TO JUST DROP A CSV FILE OR ENTER A LIST OF ADDR AND IT WILL COMPUTE THE MERKLE ROOT FOR YOU
+// TODO: ADD A DATE PICKER FOR THE START TIMESTAMP AND END TIMESTAMP
+export const CreateSaleForm: React.FC = () => {
+  const {
+    actions: { goNextStep, setSale },
+    createSaleForm: formData,
+  } = useCreateSaleState();
+  const selectedWallet = useSelectedWallet();
+  const caller = selectedWallet?.address;
+  const { handleSubmit, formState, setValue, watch } = useForm({
+    resolver: zodResolver(zodCreateSaleForm),
+    defaultValues: formData,
+  });
```

Comment:
Our forms handling is messy. We badly use `react-hook-form`.
Using `zodResolver` is a good thing. It allows to focus on building your zod objects instead of handling many props on many input components.
Don't worry if your form isn't perfect, we'll refacto later

Here is how you can modify `useForm` usage to plug correclty your form and handling errors.
Take attention to `<TCreateSaleForm>`, `mode: "all"` and `control` usage

```suggestion
  const { handleSubmit, formState, setValue, watch, control } = useForm<TCreateSaleForm>({
    resolver: zodResolver(zodCreateSaleForm),
    defaultValues: formData,
    mode: "all"
  });
```

See `TextInputCustom` usage bellow

Reply:
[af068a1](https://github.com/TERITORI/teritori-dapp/pull/1276/commits/af068a10f1b306f53c9b6170f1c46e7c846b96e7)